### PR TITLE
add draft Java API docs and link

### PR DESCRIPTION
**Pull request by nadinequinr3**

### CODE REVIEW

**Comment by yarivus** [comment]
> LGTM

@nadinequinr3 / @colin-hillis 
This is over 1000 files - that's a huge risk as a PR ...
What is the change about ?
Can we break the PRs to consumable chunks ?

**Comment by nadinequinr3** [comment]
> > LGTM
> 
> @nadinequinr3 / @colin-hillis This is over 1000 files - that's a huge risk as a PR ... What is the change about ? Can we break the PRs to consumable chunks ?

I'm adding the Javadocs, as requested by QA It's a simple copy and paste of the files generated from the code. Not sure how else to include them without doing a PR?

**Comment by yarivus** [comment]
> > > LGTM
> > 
> > 
> > @nadinequinr3 / @colin-hillis This is over 1000 files - that's a huge risk as a PR ... What is the change about ? Can we break the PRs to consumable chunks ?
> 
> I'm adding the Javadocs, as requested by QA. It's a simple copy and paste of the files generated from the code. Not sure how else to include them without doing a PR?

Doing them in multiple PRs (folder by folder) ?

**Comment by nadinequinr3** [comment]
> > > > LGTM
> > > 
> > > 
> > > @nadinequinr3 / @colin-hillis This is over 1000 files - that's a huge risk as a PR ... What is the change about ? Can we break the PRs to consumable chunks ?
> > 
> > 
> > I'm adding the Javadocs, as requested by QA. It's a simple copy and paste of the files generated from the code. Not sure how else to include them without doing a PR?
> 
> Doing them in multiple PRs (folder by folder) ?

That seems like a lot of overhead for no gain. This is not the stage at which the content within the files (which are owned by Development) is reviewed. Also, worth noting that these files are usually updated very close to a release. Do we really want to add multiple extra PRs at that late stage?

**Comment by colin-hillis** [comment]
> > LGTM
> 
> @nadinequinr3 / @colin-hillis This is over 1000 files - that's a huge risk as a PR ... What is the change about ? Can we break the PRs to consumable chunks ?

This is 100% OK. These are generated files, where the editing/approving is done pre-generation by dev, with writers reviewing at that earlier stage if necessary. Once they are generated, it is many many times more difficult and time-wasting to start breaking apart the generated files in order to facilitate a second round of reviews that are not needed anyway. Besides, even if they were broken into three parts, nobody is going to review 300-400 API files in a PR either. There is no other feasible way (without inventing work) to do this and IMO no reason to consider finding an alternative. Even if something unforeseen did break, that is exactly the point of using github. Everything can be reversed or tweaked and re-merged. However so far, there has been no need to do that for any of the big merges that we have done, so there isn't a precedent issue for worrying about this IMO.

The same comments have been made about archiving content and also creating a new version. When we create a new version based on latest iteration of previous version, that massive amount of content has already been reviewed for accuracy. Breaking it up leads to errors, much extra time and confusion, with no ACTUAL review likely to take place anyway. It is pointless. Same for archiving versions. Nobody is reviewing that many thousands of files whether broken up into one part or 10 parts and to break it up under the pretence that someone actually will review, would lead to so much extra work. It is quite literally a move to the archive, which is marked up in github as deleted from one located and created in the new location (archive). 

In summary, I think perhaps that we might be trying to solve an issue that is not an issue here. There has been no example of it being an issue in past, the content being added doesn't even need a review, and in all cases, if the worst comes to the worst, the merge can be undone.

**Comment by nadinequinr3** [comment]
Just FYI, there was a preview shared on the Slack review channel (https://nadine.preview.docs.r3.com/en/platform/corda/5.0-beta/developing/api/corda-api.html)
